### ISSUE TITLE
When tsconfig.json configures "@/*": ["src/*"] and vite.config.ts configures alias “@/”, the generated type path is incorrect.

### ISSUE THREAD

**Issue by SKT1yang** [qmhc/vite-plugin-dts#290]
### Describe the bug

I use "@/" as the path alias, and when packaging, the JS file in the product is normal. However, there is a missing "/" in the alias of the ".d.ts" file. For example, when I use "@/components/HelloWorld. vue", it should be converted to "./components/HelloWorld. vue", but in reality, it is converted to ". components/HelloWorld. vue"
<img width="1063" alt="image" src="https://github.com/qmhc/vite-plugin-dts/assets/34095649/f3f0b1a2-4008-4250-8919-9c6bcb788b64">
<img width="1103" alt="image" src="https://github.com/qmhc/vite-plugin-dts/assets/34095649/ae612840-3882-4888-a800-45c4ea134e1d">

### Reproduction

https://stackblitz.com/edit/vitejs-vite-ctrqsy?file=tsconfig.json,vite.config.ts,src%2Findex.ts 

### Steps to reproduce

1. npm run build
2. look file "dist/index.d.ts"

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.4.2 - /usr/local/bin/npm
    pnpm: 8.10.5 - /usr/local/bin/pnpm
  npmPackages:
    @vitejs/plugin-vue: ^4.5.2 => 4.5.2 
    vite: ^5.0.8 => 5.0.8 
    vite-plugin-dts: ^3.6.4 => 3.6.4
```

### Validations

- [X] Read the [FAQ](https://github.com/qmhc/vite-plugin-dts#faq).
- [X] Check that there isn't [already an issue](https://github.com/qmhc/vite-plugin-dts/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sconix** [comment]
@qmhc With this change configurations like:
`
    "paths": {
      "#*": [
        "./src/*"
      ]
    }
`

wont work anymore, not fully sure if they worked before this change did not yet have time to test with older versions. Just thought to let you already know that this fix is not perfect, will try to find time to make a new reproduction and file a new bug later.

Did quick test with the reproduction in this PR with updated dts plugin to 3.7.0 and modified it to according above setup and imports to be #components* and the output was .components/* not ./components/* (did verify to work correctly with the original @/* setup so this fixes that, but introduces new issues).

In our project with the #* configuration the plugin fails to completely replace the paths and they remain as #components/* in the output file. Have not yet figured out what is the difference with our project and the above reproduction, could be that we have multi project repo both vite and typescript config files are in subdirectory or its something else, will try to replicate this in the reproduction.

Our current setup uses vue-tsc + tsc-alias and that combination replaces the paths correctly, but would love to change to using this plugin.

**Comment by lachieh** [comment]
@sconix it's going back a while now, but this should be fixed in [4.1.0](https://github.com/qmhc/vite-plugin-dts/releases/tag/v4.1.0)
